lib/api/model/model.dart

```diff
@@ -120,6 +78,41 @@ class User {
   Map<String, dynamic> toJson() => _$UserToJson(this);
 }
 
+/// As in [User.profileData].
+@JsonSerializable(fieldRename: FieldRename.snake)
+class ProfileFieldUserData {
+  final String value;
+  final String? renderedValue;
+
+  ProfileFieldUserData({required this.value, this.renderedValue});
+
+  factory ProfileFieldUserData.fromJson(Map<String, dynamic> json) =>
+      _$ProfileFieldUserDataFromJson(json);
+
+  Map<String, dynamic> toJson() => _$ProfileFieldUserDataToJson(this);
+}
+
+/// As in [User.role].
+@JsonEnum(fieldRename: FieldRename.snake, valueField: "apiValue")
+enum UserRole{
+  owner(apiValue: 100, label: "Owner"),
+  administrator(apiValue: 200, label: "Admin"),
+  moderator(apiValue: 300, label: "Moderator"),
+  member(apiValue: 400, label: "Member"),
+  guest(apiValue: 600, label: "Guest"),
+  unknown(apiValue: null, label: "Unknown");
+
+  const UserRole({
+    required this.apiValue,
+    required this.label,
+  });
+
+  final int? apiValue;
+  final String label;
+
+  int toJson() => _$UserRoleEnumMap[this]!;
```

Comment:
I've settled on the "embrace the null" way, because even though tests pass when I remove `UserRole.toJson` we will get a crash if we ever call `User.toJson` (even with known values for the role field). I think it's a bummer we lose the original value here, but it doesn't look like there's a way to do that with the existing usage of `json_serializable` and enhanced enums. (I think there's a way to preserve it by making `UserRole` a wrapper class that saves the original value and exposes a decoded enum, do you think it's worth the effort here?)

Reply:
The other route is to drop the `unknown` here? We'd then get a parse error if the server sends us a new value. We do have other `JsonEnum`s that will similarly fail to parse if a new value shows up (e.g., `PropagateMode`, `MessageType`, `ReactionType`) although the difference with these are they are all string values (though I'm not sure that makes a difference in how likely they are to change)

Reply:
Dropping the `unknown` would also be reasonable in this case. I think it's pretty unlikely that this enum will gain a new value, just because we're in the process of converting our permissions system to a more generic architecture with arbitrary groups — so that for any further wrinkle we'd otherwise be tempted to express by adding another value here, I expect we'll instead hold out for groups and express it in those terms instead.

Though if we did want to take that route, it might be prudent to run it by a conversation in `#api design` first, to confirm that Tim agrees it's unlikely we'd want to add another value and that nobody would be sad about that becoming something that would require some compatibility engineering to do.